templates/json/inline-swift3.stencil

```diff
@@ -47,13 +47,25 @@ import Foundation
 {% macro propertyBlock key metadata data %}{% filter removeNewlines:"leading" %}
   {% set propertyName %}{{key|swiftIdentifier:"pretty"|lowerFirstWord|escapeReservedKeywords}}{% endset %}
   {% set propertyType %}{% call typeBlock metadata %}{% endset %}
-  static let {{propertyName}}: {{propertyType}} = {% call valueBlock data[key] propertyType %}
+  static let {{propertyName}}: {{propertyType}} = {% call valueBlock data[key] metadata %}
 {% endfilter %}{% endmacro %}
-{% macro valueBlock value type %}{% filter removeNewlines:"leading" %}
-  {% if type == "String" %}
+{% macro valueBlock value metadata %}{% filter removeNewlines:"leading" %}
+  {% if metadata.type == "String" %}
     "{{ value }}"
-  {% elif type|hasSuffix:"?" %}
+  {% elif metadata.type == "Optional" %}
     nil
+  {% elif metadata.type == "Array" and metadata.element.items %}
```

Comment:
I'm a bit confused. In the Fabric example above, we had the `items` key present (because heterogeneous dict), and `type` was `"Any"`… while here you use `metadata.type == "Array"`… I think i got lost somewhere in there, are heterogeneous structures supposed to be Any or Array, and if they're Any like in the Fabric example able, shouldn't that condition here be `metadata.type == "Any"` then?

Reply:
Forget for a sec about that `Any` case, I've updated the `Metadata.swift` code a bit.

Remember that the Fabric property looks like this:
```yaml
Fabric:
  APIKey: "512345678900aaafffff"
  Kits:
  - KitInfo: {}
    KitName: "Crashlytics"
```

- So the template starts with the value of "Fabric", which is a dict, so it goes into the `{% elif metadata.type == "Dictionary" %}` branch.
- For the "Kits" property, which is an array, it goes into the `{% elif metadata.type == "Array" and ... %}` branch
- For the contents of that array, which is a dictionary, it goes again into the `{% elif metadata.type == "Dictionary" %}` branch
- And so on...

Reply:
Remember that for arrays:

- type = "Array"
- element = metadata description of the Array.Element:
  - type = the type of the array element (Any/Dictionary/Array/String/....)
  - items = list of metadata description of each Array item

Reply:
See also https://github.com/SwiftGen/SwiftGen/blob/30ec6ff7fb3785e8ed8d38bab8cc5bcb3abf2583/Documentation/SwiftGenKit%20Contexts/Plist.md#example which might clarify things.

Reply:
Ohhh right Kits is an array not a dict, missed that.

In the md you linked is it normal that for Kits though we have both `element: type: Dictionary` (as in: "each element in the Kits array is a Dictionary") and `element: items: type: Dictionary` (… same?)

```yaml
              element:
                type: "Dictionary"
                items:
                - type: "Dictionary"
```

Reply:
That's simple repetition of the metadata object, yeah. It could just have well been something like:
```swift
["hello", 2, "world", ["weird": "example"]
```
Where the `element.type` = `Any`, and:
- `items.0.type` = `String`
- `items.1.type` = `Int`
- `items.2.type` = `String`
- `items.3.type` = `Dictionary`